backend/views.py

```diff
@@ -1,8 +1,36 @@
-from canvas_oauth.oauth import get_oauth_token
+import logging
+
+from canvas_oauth.oauth import get_oauth_token, handle_missing_token
+from canvas_oauth.models import CanvasOAuth2Token
+from canvas_oauth.exceptions import CanvasOAuthError, MissingTokenError, InvalidOAuthStateError, InvalidOAuthReturnError
+
+from django.contrib.auth.models import User
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render
 
+
+logger = logging.getLogger(__name__)
+
 @login_required
 def get_home_template(request):
-    access_token = get_oauth_token(request)
+    try:
+        access_token = get_oauth_token(request)
+    except InvalidOAuthStateError as err:
+        logger.error("InvalidOAuthStateError. Remove invalid refresh_token and prompt for reauthentication.")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        return handle_missing_token(request)
+    except InvalidOAuthReturnError as err:
+        logger.error("InvalidOAuthReturnError. Remove invalid refresh_token and prompt for reauthentication.")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        return handle_missing_token(request)
+    except MissingTokenError as err:
+        logger.error("MissingTokenError. Prompt for reauthentication.")
+        return handle_missing_token(request)
+    except CanvasOAuthError as err:
+        logger.error("CanvasOAuthError. Remove invalid refresh_token and prompt for reauthentication.")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        return handle_missing_token(request)
+    except Exception as err:
+        logger.error(err)
```

Comment:
I think we could just let other errors not be caught, and let the library's middleware handle them. What do you think? https://github.com/Harvard-University-iCommons/django-canvas-oauth/blob/master/canvas_oauth/middleware.py

Reply:
In summary, we could just have a `try` / `except` catching `InvalidOAuthReturnError`, and leave the rest out.

Reply:
That sounds good. I will leave the InvalidOAuthReturnError handling and remove others.